Fix Dockerfile

znerol opened the pull request:
* Define a particular release (Ubuntu trusty) of the source 
* Remove spurious dependencies.
* Perform update and install from the same RUN.
* Specify each installable package on a separate line.
* Use COPY to copy the current directory including any modifications to the image instead of pulling a fresh copy from the git repository.


The first and second change is necessary in order to make the Dockerfile functional again, the rest is in accordance to [Docker documentation](https://docs.docker.com/articles/dockerfile_best-practices/)

vecna commented:
Wonderful!! :)

vecna commented:
There are some minor issues:

  * the requirement list is changed, now they are less (python-dev libgeoip-dev  geoip-database are not more required, because geoip is no more performed by client)
  * Tor is no more a requirement, but just an option
  * is important having something that update the media list, because if someone create the docker image now, and the media list is updated, how can get notice and download automatically the new file ? 

what do you think about ?

znerol commented:
> python-dev libgeoip-dev geoip-database are not more required, because geoip is no more performed by client.

Ok, also removed `gcc` (because that is installed anyway via `python-pip` which recommends `build-essential`

> Tor is no more a requirement, but just an option.

Also removed that because it is not functional anyway with the current `Dockerfile`. In order to make it work, it would be necessary to manually start `tor` from a script executed on `ENTRYPOINT`. This might look very similar to the one included with the [official postgres docker image](https://github.com/docker-library/postgres).

> is important having something that update the media list, because if someone create the docker image now, and the media list is updated, how can get notice and download automatically the new file ?

From what I understand, docker images are not intended to be updated at all. Containers are supposed to be stateless and application data is expected to be kept outside of the image (e.g. with volumes and data images). Whenever either the application code or the base system needs an update, the whole image is rebuilt and a new container instance is deployed. It is even possible to set up automated builds for repositories on github, see [this guide](http://docs.docker.com/docker-hub/builds/)
